mono/metadata/threads.c

```diff
@@ -1886,18 +1886,23 @@ ves_icall_System_Threading_Thread_GetName_internal (MonoInternalThreadHandle thr
 void 
 mono_thread_set_name_internal (MonoInternalThread *this_obj, MonoString *name, gboolean permanent, gboolean reset, MonoError *error)
 {
+	// Mostly setting thread name is best-effort, and failure does not matter,
+	// but it is also a public API with a documented failure mode.
+
 	MonoNativeThreadId tid = 0;
 
-	LOCK_THREAD (this_obj);
+	if (error)
```

Comment:
I weakly-dislike this.  We don't use a NULL MonoError as a sentinel value anywhere else in the runtime. 

What's the savings; we avoid one call to `mono_error_set_*`?

Reply:
If the error is going to be ignored, it seems better to avoid preparing it, and possibly that failing.
But I guess it could go either way, in terms of, when debugging it, if you want to hit mono_error_prepare or not.
The point is to avoid failability and noise, not cycles per se.
Debuggability is subtle and rarely considered and I not necessarily good at it.